README.md
Yii2 PDFJS
==========
[![Latest Stable Version](https://poser.pugx.org/yii2assets/yii2-pdfjs/v/stable)](https://packagist.org/packages/yii2assets/yii2-pdfjs) [![Total Downloads](https://poser.pugx.org/yii2assets/yii2-pdfjs/downloads)](https://packagist.org/packages/yii2assets/yii2-pdfjs) [![Latest Unstable Version](https://poser.pugx.org/yii2assets/yii2-pdfjs/v/unstable)](https://packagist.org/packages/yii2assets/yii2-pdfjs) [![License](https://poser.pugx.org/yii2assets/yii2-pdfjs/license)](https://packagist.org/packages/yii2assets/yii2-pdfjs) [![composer.lock](https://poser.pugx.org/yii2assets/yii2-pdfjs/composerlock)](https://packagist.org/packages/yii2assets/yii2-pdfjs)

Yii2 PDFJS bundle of [PDF.js](https://mozilla.github.io/pdf.js/) plugin. PDF.js Portable Document Format (PDF) viewer.

Installation
------------

The preferred way to install this extension is through [composer](http://getcomposer.org/download/).

Either run

```
php composer require --prefer-dist dzhemile/yii2-pdfjs ">=1.0"
```

or add

```
"dzhemile/yii2-pdfjs": ">=1.0"
```

to the require section of your `composer.json` file.

Module Setup
-----
The extension has been created as a module to enable access preview `pdf` file. you must add the module to your configuration file.

```php
//....

'modules'=>[
  'pdfjs' => [
       'class' => '\dzhemile\pdfjs\Module',
   ],
],

//...
```



Usage
-----

Once the extension is installed, simply use it in your code by  :

```php
<?php
use yii\helpers\Url;
?>

<?= \dzhemile\pdfjs\PdfJs::widget([
  'url'=> Url::base().'/downloads/manualStart_up.pdf'
]); ?>
```
![](docs/images/pdfjs1.png)

Using a `yii2-pdfjs` widget inside a modal dialog.
----

```php
<?php
use yii\bootstrap\Modal;
use yii\helpers\Url;

Modal::begin([
    'header' => '<h2>Hello world</h2>',
    'toggleButton' => ['label' => 'click me'],
]);

echo \dzhemile\pdfjs\PdfJs::widget([
  'url' => Url::base().'/downloads/manualStart_up.pdf'
]);

Modal::end();
?>
```
![](docs/images/pdfjs2.png)

Using a `yii2-pdfjs` widget fullscreen viewer
-----

```
http://app-url/index.php?r=pdfjs&file=download/manualStart_up.pdf
```
![](docs/images/pdfjs3.png)


Config Width & Height
-----
```php
<?php
use yii\helpers\Url;
?>

<?= \dzhemile\pdfjs\PdfJs::widget([
  'width'=>'100%',
  'height'=> '500px',
  'url'=> Url::base().'/downloads/pdfjs.pdf'
]);
?>

```

Config disable toolbar buttons
-------------------------------

```php
<?php
use yii\helpers\Url;
?>

<?= \dzhemile\pdfjs\PdfJs::widget([
  'url'=> Url::base().'/downloads/pdfjs.pdf',
  'buttons'=>[
    'presentationMode' => false,
    'openFile' => false,
    'print' => true,
    'download' => true,
    'viewBookmark' => false,
    'secondaryToolbarToggle' => false
  ]
]);
?>
```

![](docs/images/pdfjs4.png)

Add Watermark text
------------------
```php
//....

'modules'=>[
  'pdfjs' => [
       'class' => '\dzhemile\pdfjs\Module',
       'waterMark'=>[
         'text'=>' Yii2 PDF.JS',
         'color'=> 'red',
         'alpha'=>'0.3'
       ]
   ],
],

//...
```

![](docs/images/pdfjs5.png)


CHANGELOG.md
Yii2 PDFJS Change Log
=================================



Version 1.1 August 14, 2023
---------------------------

- Updated PDFJS to version 3.9.179


Version 1.0.2 July 22, 2016
---------------------

- (Enh #14): Added watermark text support, update css background-color iframe


Version 1.0.1 July 21, 2016
---------------------

- (Enh #8): Added option buttons config


PdfJs.php
<?php

namespace dzhemile\pdfjs;
use Yii;
use yii\helpers\Html;
use yii\helpers\Url;
use yii\helpers\ArrayHelper;

/**
 * PdfJs Widget
 * @author Sathit Seethaphon <dixonsatit@gmail.com>
 */
class PdfJs extends \yii\base\Widget
{
    public $url=null;

    public $width = '100%';

    public $height = '500px';

    public $options = [];

    public $buttons = [];

    public function init()
    {
      parent::init();
      $buttons = Yii::$app->getModule('pdfjs')->buttons;
      $this->buttons = array_merge($buttons, $this->buttons);
      PdfJsAsset::register($this->getView());
    }

    public function run()
    {
      if(!array_key_exists('style',$this->options)){
        $this->options['style'] = 'border:solid 2px #404040; width:'.$this->width.'; height:'.$this->height.';';
      }
      return $this->render('viewer',[
        'options' => $this->options,
        'url' => $this->url,
        'buttons'=>$this->buttons,
        'id'=>$this->id
      ]);
    }
}


Module.php
<?php

namespace dzhemile\pdfjs;

/**
 * pdfjs module definition class
 * @author Sathit Seethaphon <dixonsatit@gmail.com>
 */
class Module extends \yii\base\Module
{
    public $buttons = [];
    public $waterMark = false;

    /**
     * @inheritdoc
     */
    public $controllerNamespace = 'dzhemile\pdfjs\controllers';

    /**
     * @inheritdoc
     */
    public function init()
    {
        parent::init();

        $waterMarkDefault = [
          'text'=>'',
          'alpha'=>'0.5',
          'color'=>'red'
        ];

        if($this->waterMark!==false && is_array($this->waterMark)){
          $this->waterMark = array_merge($waterMarkDefault,$this->waterMark);
        }else{
          $this->waterMark = $waterMarkDefault;
        }


        $this->buttons = array_merge([
          'presentationMode' => true,
          'openFile' => true,
          'print' => true,
          'download' => true,
          'viewBookmark' => true,
          'secondaryToolbarToggle'=> true
        ], $this->buttons);

        // custom initialization code goes here
    }
}


views/layouts/main.php
<?php
/* @var $this \yii\web\View */
/* @var $content string */
use yii\helpers\Html;
use yii\web\View;
use yii\helpers\Url;
use dzhemile\pdfjs\PdfJsAsset;
$bundle = PdfJsAsset::register($this);
?>
<?php $this->beginPage() ?>
<!DOCTYPE html>
<!--
Copyright 2012 Mozilla Foundation

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Adobe CMap resources are covered by their own copyright but the same license:

    Copyright 1990-2015 Adobe Systems Incorporated.

See https://github.com/adobe-type-tools/cmap-resources
-->
<html dir="ltr" mozdisallowselectionprint">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="google" content="notranslate">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>PDF.js viewer</title>
    <!-- This snippet is used in production (included from viewer.html) -->
    <link rel="resource" type="application/l10n" href="<?=$bundle->baseUrl?>/web/locale/locale.properties"/>
    <script type="text/javascript">
    var WORKER_URL = "<?=$bundle->baseUrl?>/build/pdf.worker.js";
    </script>
<?php $this->head() ?>
</head>
<body tabindex="1">
<?php $this->beginBody() ?>
  <?=$content?>
<?php $this->endBody() ?>
</body>
</html>
<?php $this->endPage() ?>


controllers/DefaultController.php
<?php

namespace dzhemile\pdfjs\controllers;

use Yii;
use yii\web\Controller;

/**
 * Default controller for the `pdfjs` module
 * @author Sathit Seethaphon <dixonsatit@gmail.com>
 */
class DefaultController extends Controller
{
    public $layout = 'main';
    /**
     * Renders the index view for the module
     * @return string
     */
    public function actionIndex()
    {
        $buttons = Yii::$app->getModule('pdfjs')->buttons;
        $waterMark = Yii::$app->getModule('pdfjs')->waterMark;
        if(Yii::$app->request->getIsPost()){

          $widgitButtonConfig =  Yii::$app->request->post();
          if(isset(Yii::$app->request->csrfParam)){
            unset($widgitButtonConfig[Yii::$app->request->csrfParam]);
          }

          foreach ($widgitButtonConfig as $key => $value) {
            $widgitButtonConfig[$key] = trim($value) == '0' ? false:true;
          }
          $buttons = array_merge($buttons,$widgitButtonConfig);
        }

        return $this->render('index',[
          'buttons'=>$buttons,
          'waterMark'=>$waterMark
        ]);
    }
}


views/default/index.php
<?php
use yii\web\View;

$js = '';
foreach ($buttons as $btn => $value) {
  if($value === false){
    $js .= "$('#{$btn}').hide();\n";
  }
}

$pdjsConfig = "
var WATERMARK_ALPHA = '".$waterMark['alpha']."';
var WATERMARK_TEXT = '".$waterMark['text']."';
var WATERMARK_TEXT_COLOR = '".$waterMark['color']."';
";
$this->registerJs($pdjsConfig,View::POS_HEAD);
$this->registerJs($js);
?>
<div id="outerContainer">

<div id="sidebarContainer">
  <div id="toolbarSidebar">
    <div id="toolbarSidebarLeft">
      <div id="sidebarViewButtons" class="splitToolbarButton toggled" role="radiogroup">
        <button id="viewThumbnail" class="toolbarButton toggled" title="Show Thumbnails" tabindex="2" data-l10n-id="thumbs" role="radio" aria-checked="true" aria-controls="thumbnailView">
           <span data-l10n-id="thumbs_label">Thumbnails</span>
        </button>
        <button id="viewOutline" class="toolbarButton" title="Show Document Outline (double-click to expand/collapse all items)" tabindex="3" data-l10n-id="document_outline" role="radio" aria-checked="false" aria-controls="outlineView">
           <span data-l10n-id="document_outline_label">Document Outline</span>
        </button>
        <button id="viewAttachments" class="toolbarButton" title="Show Attachments" tabindex="4" data-l10n-id="attachments" role="radio" aria-checked="false" aria-controls="attachmentsView">
           <span data-l10n-id="attachments_label">Attachments</span>
        </button>
        <button id="viewLayers" class="toolbarButton" title="Show Layers (double-click to reset all layers to the default state)" tabindex="5" data-l10n-id="layers" role="radio" aria-checked="false" aria-controls="layersView">
           <span data-l10n-id="layers_label">Layers</span>
        </button>
      </div>
    </div>

    <div id="toolbarSidebarRight">
      <div id="outlineOptionsContainer" class="hidden">
        <div class="verticalToolbarSeparator"></div>

        <button id="currentOutlineItem" class="toolbarButton" disabled="disabled" title="Find Current Outline Item" tabindex="6" data-l10n-id="current_outline_item">
          <span data-l10n-id="current_outline_item_label">Current Outline Item</span>
        </button>
      </div>
    </div>
  </div>
  <div id="sidebarContent">
    <div id="thumbnailView">
    </div>
    <div id="outlineView" class="hidden">
    </div>
    <div id="attachmentsView" class="hidden">
    </div>
    <div id="layersView" class="hidden">
    </div>
  </div>
  <div id="sidebarResizer"></div>
</div>  <!-- sidebarContainer -->

<div id="mainContainer">
  <div class="findbar hidden doorHanger" id="findbar">
    <div id="findbarInputContainer">
      <input id="findInput" class="toolbarField" title="Find" placeholder="Find in document…" tabindex="91" data-l10n-id="find_input" aria-invalid="false">
      <div class="splitToolbarButton">
        <button id="findPrevious" class="toolbarButton" title="Find the previous occurrence of the phrase" tabindex="92" data-l10n-id="find_previous">
          <span data-l10n-id="find_previous_label">Previous</span>
        </button>
        <div class="splitToolbarButtonSeparator"></div>
        <button id="findNext" class="toolbarButton" title="Find the next occurrence of the phrase" tabindex="93" data-l10n-id="find_next">
          <span data-l10n-id="find_next_label">Next</span>
        </button>
      </div>
    </div>

    <div id="findbarOptionsOneContainer">
      <input type="checkbox" id="findHighlightAll" class="toolbarField" tabindex="94">
      <label for="findHighlightAll" class="toolbarLabel" data-l10n-id="find_highlight">Highlight All</label>
      <input type="checkbox" id="findMatchCase" class="toolbarField" tabindex="95">
      <label for="findMatchCase" class="toolbarLabel" data-l10n-id="find_match_case_label">Match Case</label>
    </div>
    <div id="findbarOptionsTwoContainer">
      <input type="checkbox" id="findMatchDiacritics" class="toolbarField" tabindex="96">
      <label for="findMatchDiacritics" class="toolbarLabel" data-l10n-id="find_match_diacritics_label">Match Diacritics</label>
      <input type="checkbox" id="findEntireWord" class="toolbarField" tabindex="97">
      <label for="findEntireWord" class="toolbarLabel" data-l10n-id="find_entire_word_label">Whole Words</label>
    </div>

    <div id="findbarMessageContainer" aria-live="polite">
      <span id="findResultsCount" class="toolbarLabel"></span>
      <span id="findMsg" class="toolbarLabel"></span>
    </div>
  </div>  <!-- findbar -->

  <div class="editorParamsToolbar hidden doorHangerRight" id="editorFreeTextParamsToolbar">
    <div class="editorParamsToolbarContainer">
      <div class="editorParamsSetter">
        <label for="editorFreeTextColor" class="editorParamsLabel" data-l10n-id="editor_free_text_color">Color</label>
        <input type="color" id="editorFreeTextColor" class="editorParamsColor" tabindex="100">
      </div>
      <div class="editorParamsSetter">
        <label for="editorFreeTextFontSize" class="editorParamsLabel" data-l10n-id="editor_free_text_size">Size</label>
        <input type="range" id="editorFreeTextFontSize" class="editorParamsSlider" value="10" min="5" max="100" step="1" tabindex="101">
      </div>
    </div>
  </div>

  <div class="editorParamsToolbar hidden doorHangerRight" id="editorInkParamsToolbar">
    <div class="editorParamsToolbarContainer">
      <div class="editorParamsSetter">
        <label for="editorInkColor" class="editorParamsLabel" data-l10n-id="editor_ink_color">Color</label>
        <input type="color" id="editorInkColor" class="editorParamsColor" tabindex="102">
      </div>
      <div class="editorParamsSetter">
        <label for="editorInkThickness" class="editorParamsLabel" data-l10n-id="editor_ink_thickness">Thickness</label>
        <input type="range" id="editorInkThickness" class="editorParamsSlider" value="1" min="1" max="20" step="1" tabindex="103">
      </div>
      <div class="editorParamsSetter">
        <label for="editorInkOpacity" class="editorParamsLabel" data-l10n-id="editor_ink_opacity">Opacity</label>
        <input type="range" id="editorInkOpacity" class="editorParamsSlider" value="100" min="1" max="100" step="1" tabindex="104">
      </div>
    </div>
  </div>

  <div id="secondaryToolbar" class="secondaryToolbar hidden doorHangerRight">
    <div id="secondaryToolbarButtonContainer">
      <button id="secondaryOpenFile" class="secondaryToolbarButton visibleLargeView" title="Open File" tabindex="51" data-l10n-id="open_file">
        <span data-l10n-id="open_file_label">Open</span>
      </button>

      <button id="secondaryPrint" class="secondaryToolbarButton visibleMediumView" title="Print" tabindex="52" data-l10n-id="print">
        <span data-l10n-id="print_label">Print</span>
      </button>

      <button id="secondaryDownload" class="secondaryToolbarButton visibleMediumView" title="Save" tabindex="53" data-l10n-id="save">
        <span data-l10n-id="save_label">Save</span>
      </button>

      <div class="horizontalToolbarSeparator visibleLargeView"></div>

      <button id="presentationMode" class="secondaryToolbarButton" title="Switch to Presentation Mode" tabindex="54" data-l10n-id="presentation_mode">
        <span data-l10n-id="presentation_mode_label">Presentation Mode</span>
      </button>

      <a href="#" id="viewBookmark" class="secondaryToolbarButton" title="Current Page (View URL from Current Page)" tabindex="55" data-l10n-id="bookmark1">
        <span data-l10n-id="bookmark1_label">Current Page</span>
      </a>

      <div id="viewBookmarkSeparator" class="horizontalToolbarSeparator"></div>

      <button id="firstPage" class="secondaryToolbarButton" title="Go to First Page" tabindex="56" data-l10n-id="first_page">
        <span data-l10n-id="first_page_label">Go to First Page</span>
      </button>
      <button id="lastPage" class="secondaryToolbarButton" title="Go to Last Page" tabindex="57" data-l10n-id="last_page">
        <span data-l10n-id="last_page_label">Go to Last Page</span>
      </button>

      <div class="horizontalToolbarSeparator"></div>

      <button id="pageRotateCw" class="secondaryToolbarButton" title="Rotate Clockwise" tabindex="58" data-l10n-id="page_rotate_cw">
        <span data-l10n-id="page_rotate_cw_label">Rotate Clockwise</span>
      </button>
      <button id="pageRotateCcw" class="secondaryToolbarButton" title="Rotate Counterclockwise" tabindex="59" data-l10n-id="page_rotate_ccw">
        <span data-l10n-id="page_rotate_ccw_label">Rotate Counterclockwise</span>
      </button>

      <div class="horizontalToolbarSeparator"></div>

      <div id="cursorToolButtons" role="radiogroup">
        <button id="cursorSelectTool" class="secondaryToolbarButton toggled" title="Enable Text Selection Tool" tabindex="60" data-l10n-id="cursor_text_select_tool" role="radio" aria-checked="true">
          <span data-l10n-id="cursor_text_select_tool_label">Text Selection Tool</span>
        </button>
        <button id="cursorHandTool" class="secondaryToolbarButton" title="Enable Hand Tool" tabindex="61" data-l10n-id="cursor_hand_tool" role="radio" aria-checked="false">
          <span data-l10n-id="cursor_hand_tool_label">Hand Tool</span>
        </button>
      </div>

      <div class="horizontalToolbarSeparator"></div>

      <div id="scrollModeButtons" role="radiogroup">
        <button id="scrollPage" class="secondaryToolbarButton" title="Use Page Scrolling" tabindex="62" data-l10n-id="scroll_page" role="radio" aria-checked="false">
          <span data-l10n-id="scroll_page_label">Page Scrolling</span>
        </button>
        <button id="scrollVertical" class="secondaryToolbarButton toggled" title="Use Vertical Scrolling" tabindex="63" data-l10n-id="scroll_vertical" role="radio" aria-checked="true">
          <span data-l10n-id="scroll_vertical_label" >Vertical Scrolling</span>
        </button>
        <button id="scrollHorizontal" class="secondaryToolbarButton" title="Use Horizontal Scrolling" tabindex="64" data-l10n-id="scroll_horizontal" role="radio" aria-checked="false">
          <span data-l10n-id="scroll_horizontal_label">Horizontal Scrolling</span>
        </button>
        <button id="scrollWrapped" class="secondaryToolbarButton" title="Use Wrapped Scrolling" tabindex="65" data-l10n-id="scroll_wrapped" role="radio" aria-checked="false">
          <span data-l10n-id="scroll_wrapped_label">Wrapped Scrolling</span>
        </button>
      </div>

      <div class="horizontalToolbarSeparator"></div>

      <div id="spreadModeButtons" role="radiogroup">
        <button id="spreadNone" class="secondaryToolbarButton toggled" title="Do not join page spreads" tabindex="66" data-l10n-id="spread_none" role="radio" aria-checked="true">
          <span data-l10n-id="spread_none_label">No Spreads</span>
        </button>
        <button id="spreadOdd" class="secondaryToolbarButton" title="Join page spreads starting with odd-numbered pages" tabindex="67" data-l10n-id="spread_odd" role="radio" aria-checked="false">
          <span data-l10n-id="spread_odd_label">Odd Spreads</span>
        </button>
        <button id="spreadEven" class="secondaryToolbarButton" title="Join page spreads starting with even-numbered pages" tabindex="68" data-l10n-id="spread_even" role="radio" aria-checked="false">
          <span data-l10n-id="spread_even_label">Even Spreads</span>
        </button>
      </div>

      <div class="horizontalToolbarSeparator"></div>

      <button id="documentProperties" class="secondaryToolbarButton" title="Document Properties…" tabindex="69" data-l10n-id="document_properties" aria-controls="documentPropertiesDialog">
        <span data-l10n-id="document_properties_label">Document Properties…</span>
      </button>
    </div>
  </div>  <!-- secondaryToolbar -->

  <div class="toolbar">
    <div id="toolbarContainer">
      <div id="toolbarViewer">
        <div id="toolbarViewerLeft">
          <button id="sidebarToggle" class="toolbarButton" title="Toggle Sidebar" tabindex="11" data-l10n-id="toggle_sidebar" aria-expanded="false" aria-controls="sidebarContainer">
            <span data-l10n-id="toggle_sidebar_label">Toggle Sidebar</span>
          </button>
          <div class="toolbarButtonSpacer"></div>
          <button id="viewFind" class="toolbarButton" title="Find in Document" tabindex="12" data-l10n-id="findbar" aria-expanded="false" aria-controls="findbar">
            <span data-l10n-id="findbar_label">Find</span>
          </button>
          <div class="splitToolbarButton hiddenSmallView">
            <button class="toolbarButton" title="Previous Page" id="previous" tabindex="13" data-l10n-id="previous">
              <span data-l10n-id="previous_label">Previous</span>
            </button>
            <div class="splitToolbarButtonSeparator"></div>
            <button class="toolbarButton" title="Next Page" id="next" tabindex="14" data-l10n-id="next">
              <span data-l10n-id="next_label">Next</span>
            </button>
          </div>
          <input type="number" id="pageNumber" class="toolbarField" title="Page" value="1" min="1" tabindex="15" data-l10n-id="page" autocomplete="off">
          <span id="numPages" class="toolbarLabel"></span>
        </div>
        <div id="toolbarViewerRight">
          <button id="openFile" class="toolbarButton hiddenLargeView" title="Open File" tabindex="31" data-l10n-id="open_file">
            <span data-l10n-id="open_file_label">Open</span>
          </button>

          <button id="print" class="toolbarButton hiddenMediumView" title="Print" tabindex="32" data-l10n-id="print">
            <span data-l10n-id="print_label">Print</span>
          </button>

          <button id="download" class="toolbarButton hiddenMediumView" title="Save" tabindex="33" data-l10n-id="save">
            <span data-l10n-id="save_label">Save</span>
          </button>

          <div class="verticalToolbarSeparator hiddenMediumView"></div>

          <div id="editorModeButtons" class="splitToolbarButton toggled" role="radiogroup">
            <button id="editorStamp" class="toolbarButton hidden" disabled="disabled" title="Image" role="radio" aria-checked="false" tabindex="34" data-l10n-id="editor_stamp">
              <span data-l10n-id="editor_stamp_label">Image</span>
            </button>
            <button id="editorFreeText" class="toolbarButton" disabled="disabled" title="Text" role="radio" aria-checked="false" aria-controls="editorFreeTextParamsToolbar" tabindex="35" data-l10n-id="editor_free_text2">
              <span data-l10n-id="editor_free_text2_label">Text</span>
            </button>
            <button id="editorInk" class="toolbarButton" disabled="disabled" title="Draw" role="radio" aria-checked="false" aria-controls="editorInkParamsToolbar" tabindex="36" data-l10n-id="editor_ink2">
              <span data-l10n-id="editor_ink2_label">Draw</span>
            </button>
          </div>

          <div id="editorModeSeparator" class="verticalToolbarSeparator"></div>

          <button id="secondaryToolbarToggle" class="toolbarButton" title="Tools" tabindex="48" data-l10n-id="tools" aria-expanded="false" aria-controls="secondaryToolbar">
            <span data-l10n-id="tools_label">Tools</span>
          </button>
        </div>
        <div id="toolbarViewerMiddle">
          <div class="splitToolbarButton">
            <button id="zoomOut" class="toolbarButton" title="Zoom Out" tabindex="21" data-l10n-id="zoom_out">
              <span data-l10n-id="zoom_out_label">Zoom Out</span>
            </button>
            <div class="splitToolbarButtonSeparator"></div>
            <button id="zoomIn" class="toolbarButton" title="Zoom In" tabindex="22" data-l10n-id="zoom_in">
              <span data-l10n-id="zoom_in_label">Zoom In</span>
             </button>
          </div>
          <span id="scaleSelectContainer" class="dropdownToolbarButton">
            <select id="scaleSelect" title="Zoom" tabindex="23" data-l10n-id="zoom">
              <option id="pageAutoOption" title="" value="auto" selected="selected" data-l10n-id="page_scale_auto">Automatic Zoom</option>
              <option id="pageActualOption" title="" value="page-actual" data-l10n-id="page_scale_actual">Actual Size</option>
              <option id="pageFitOption" title="" value="page-fit" data-l10n-id="page_scale_fit">Page Fit</option>
              <option id="pageWidthOption" title="" value="page-width" data-l10n-id="page_scale_width">Page Width</option>
              <option id="customScaleOption" title="" value="custom" disabled="disabled" hidden="true"></option>
              <option title="" value="0.5" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 50 }'>50%</option>
              <option title="" value="0.75" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 75 }'>75%</option>
              <option title="" value="1" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 100 }'>100%</option>
              <option title="" value="1.25" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 125 }'>125%</option>
              <option title="" value="1.5" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 150 }'>150%</option>
              <option title="" value="2" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 200 }'>200%</option>
              <option title="" value="3" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 300 }'>300%</option>
              <option title="" value="4" data-l10n-id="page_scale_percent" data-l10n-args='{ "scale": 400 }'>400%</option>
            </select>
          </span>
        </div>
      </div>
      <div id="loadingBar">
        <div class="progress">
          <div class="glimmer">
          </div>
        </div>
      </div>
    </div>
  </div>

  <div id="viewerContainer" tabindex="0">
    <div id="viewer" class="pdfViewer"></div>
  </div>
</div> <!-- mainContainer -->

<div id="dialogContainer">
  <dialog id="passwordDialog">
    <div class="row">
      <label for="password" id="passwordText" data-l10n-id="password_label">Enter the password to open this PDF file:</label>
    </div>
    <div class="row">
      <input type="password" id="password" class="toolbarField">
    </div>
    <div class="buttonRow">
      <button id="passwordCancel" class="dialogButton"><span data-l10n-id="password_cancel">Cancel</span></button>
      <button id="passwordSubmit" class="dialogButton"><span data-l10n-id="password_ok">OK</span></button>
    </div>
  </dialog>
  <dialog id="documentPropertiesDialog">
    <div class="row">
      <span id="fileNameLabel" data-l10n-id="document_properties_file_name">File name:</span>
      <p id="fileNameField" aria-labelledby="fileNameLabel">-</p>
    </div>
    <div class="row">
      <span id="fileSizeLabel" data-l10n-id="document_properties_file_size">File size:</span>
      <p id="fileSizeField" aria-labelledby="fileSizeLabel">-</p>
    </div>
    <div class="separator"></div>
    <div class="row">
      <span id="titleLabel" data-l10n-id="document_properties_title">Title:</span>
      <p id="titleField" aria-labelledby="titleLabel">-</p>
    </div>
    <div class="row">
      <span id="authorLabel" data-l10n-id="document_properties_author">Author:</span>
      <p id="authorField" aria-labelledby="authorLabel">-</p>
    </div>
    <div class="row">
      <span id="subjectLabel" data-l10n-id="document_properties_subject">Subject:</span>
      <p id="subjectField" aria-labelledby="subjectLabel">-</p>
    </div>
    <div class="row">
      <span id="keywordsLabel" data-l10n-id="document_properties_keywords">Keywords:</span>
      <p id="keywordsField" aria-labelledby="keywordsLabel">-</p>
    </div>
    <div class="row">
      <span id="creationDateLabel" data-l10n-id="document_properties_creation_date">Creation Date:</span>
      <p id="creationDateField" aria-labelledby="creationDateLabel">-</p>
    </div>
    <div class="row">
      <span id="modificationDateLabel" data-l10n-id="document_properties_modification_date">Modification Date:</span>
      <p id="modificationDateField" aria-labelledby="modificationDateLabel">-</p>
    </div>
    <div class="row">
      <span id="creatorLabel" data-l10n-id="document_properties_creator">Creator:</span>
      <p id="creatorField" aria-labelledby="creatorLabel">-</p>
    </div>
    <div class="separator"></div>
    <div class="row">
      <span id="producerLabel" data-l10n-id="document_properties_producer">PDF Producer:</span>
      <p id="producerField" aria-labelledby="producerLabel">-</p>
    </div>
    <div class="row">
      <span id="versionLabel" data-l10n-id="document_properties_version">PDF Version:</span>
      <p id="versionField" aria-labelledby="versionLabel">-</p>
    </div>
    <div class="row">
      <span id="pageCountLabel" data-l10n-id="document_properties_page_count">Page Count:</span>
      <p id="pageCountField" aria-labelledby="pageCountLabel">-</p>
    </div>
    <div class="row">
      <span id="pageSizeLabel" data-l10n-id="document_properties_page_size">Page Size:</span>
      <p id="pageSizeField" aria-labelledby="pageSizeLabel">-</p>
    </div>
    <div class="separator"></div>
    <div class="row">
      <span id="linearizedLabel" data-l10n-id="document_properties_linearized">Fast Web View:</span>
      <p id="linearizedField" aria-labelledby="linearizedLabel">-</p>
    </div>
    <div class="buttonRow">
      <button id="documentPropertiesClose" class="dialogButton"><span data-l10n-id="document_properties_close">Close</span></button>
    </div>
  </dialog>
  <dialog id="printServiceDialog" style="min-width: 200px;">
    <div class="row">
      <span data-l10n-id="print_progress_message">Preparing document for printing…</span>
    </div>
    <div class="row">
      <progress value="0" max="100"></progress>
      <span data-l10n-id="print_progress_percent" data-l10n-args='{ "progress": 0 }' class="relative-progress">0%</span>
    </div>
    <div class="buttonRow">
      <button id="printCancel" class="dialogButton"><span data-l10n-id="print_progress_close">Cancel</span></button>
    </div>
  </dialog>
</div>  <!-- dialogContainer -->

</div> <!-- outerContainer -->
<div id="printContainer"></div>

<input type="file" id="fileInput" class="hidden">
